### ISSUE TITLE
Add MVP architecture pattern

### ISSUE THREAD

**Issue by rajtheinnovator** [rajtheinnovator/ArchitecturePatterns#1]
Add a basic implementation of MVP pattern

**Comment by drcabral** [comment]
This one is still open? I saw that in the code already has folders related to this pattern

**Comment by rajtheinnovator** [comment]
@drcabral yes, it's still open. What I've done so far is implementing very basic single activity example with hard coded strings as data. Basically, I want to implement it with real use example like in the article on the following example. This is obviously for learning and so I guess, the more closer I can get it to typical use case, better it'll for anyone coming for seeing a practical example.Though we can also change the api if marvel api doesn't look good: https://android.jlelse.eu/avenging-android-mvp-23461aebe9b5

**Comment by drcabral** [comment]
Hey @rajtheinnovator , I will work on this today, so, as soon as possible i'll send the PR.

**Comment by rajtheinnovator** [comment]
Sure @drcabral That'd be a nice thing. And more helpful, if you can add proper Javadoc and references used.
